tests/Microsoft.Identity.Test.Unit.net45/AppConfigTests/PublicClientApplicationBuilderTests.cs

```diff
@@ -354,5 +355,44 @@ public void TestAadAuthorityWithInvalidSegmentCount()
                 Assert.AreEqual(CoreErrorMessages.AuthorityDoesNotHaveTwoSegments, ex.Message);
             }
         }
+
+        [TestMethod]
+        public void MatsAndTelemetryCallbackCannotBothBeConfiguredAtTheSameTime()
+        {
+            try
+            {
+                var app = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                    .WithTelemetry((List<Dictionary<string, string>> events) => {})
+                    .WithMatsTelemetry(new MatsConfig())
+                    .Build();
+                Assert.Fail("Should not reach here, exception should be thrown");
+            }
+            catch (Exception ex)
+            {
+                Assert.IsTrue(ex is InvalidOperationException);
+                Assert.AreEqual(CoreErrorMessages.MatsAndTelemetryCallbackCannotBeConfiguredSimultaneously, ex.Message);
+            }
+        }
+
+        [TestMethod]
+        public void MatsCanBeProperlyConfigured()
+        {
+            var matsConfig = new MatsConfig
+            {
+                AppName = "some app name",
+                AppVer = "some app version",
+                SessionId = "some session id"
+            };
+
+            var app = PublicClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                .WithMatsTelemetry(matsConfig)
```

Comment:
I vaguely remember that we already have telemetry hooks for setting the AppName and the AppVersion. Can those be reused or set as well?